jetty-documentation/src/main/asciidoc/reference/troubleshooting/troubleshooting-locked-files.adoc

```diff
@@ -26,80 +26,53 @@ Effectively this means that you have to stop Jetty to update a file.
 
 [source, xml, subs="{sub-order}"]
 ----
-<init-param>
-   <param-name>useFileMappedBuffer</param-name>
-   <param-value>true</param-value> <!-- change to false -->
- </init-param>
- 
-    
+ <servlet>
+   <servlet-name>default</servlet-name>
+   <init-param>
+     <param-name>useFileMappedBuffer</param-name>
+     <param-value>false</param-value>
+   </init-param>
+ </servlet>
 ----
 
-Make sure to apply your custom `webdefault.xml` file to all of your webapps.
-You can do that by changing the configuration of the Deployment Manager in `etc/jetty-deploy.xml`.
+===== Using a context xml file
+
+You can create or update a context xml file that configures your webapp to apply the setting to disable memory-mapped file buffers.
+Add the following to your context xml file:
 
 [source, xml, subs="{sub-order}"]
 ----
-<Call id="webappprovider" name="addAppProvider">
-  <Arg>
-    <New class="org.eclipse.jetty.deploy.providers.WebAppProvider">
-      .
-      .
-      <!-- this should be the new custom webdefault.xml or change should be made in this file -->
-      <Set name="defaultsDescriptor"><Property name="jetty.home" default="." />/etc/webdefault.xml</Set>
-      <Set name="scanInterval">1</Set>
-      <Set name="extractWars">true</Set>
-      .
-      .
-    </New>
-  </Arg>
-</Call>
-
-    
+  <Call name="setInitParameter">
+      <Arg>org.eclipse.jetty.servlet.Default.useFileMappedBuffer</Arg>
+      <Arg>false</Arg>
+  </Call>
 ----
 
-Alternatively, if you have individually configured your webapps with context xml files, you need to call the `WebAppContext.setDefaultsDescriptor(String path)` method:
 
-[source, xml, subs="{sub-order}"]
-----
-<New id="myWebAppContext"  class="org.eclipse.jetty.webapp.WebAppContext">
-  <Set name="contextPath">/</Set>
-  <Set name="war">./webapps/fredapp</Set>
-  <Set name="defaultsDescriptor">/home/fred/jetty/mywebdefaults.xml</Set>
-  .
-  .
-</New>
-
-    
-----
+===== Using the jetty maven plugin
 
-Instead, you could redefine the DefaultServlet in your web.xml file, making sure to set useFileMappedBuffer to false:
+If you don't want to use either of the other two solutions, you can configure the plugin directly to disable memory-mapped file buffers.
+Add the following to the plugin's configuration under the <webApp> element:
 
 [source, xml, subs="{sub-order}"]
 ----
-<web-app ...>
- ...
- <servlet>
-     <servlet-name>default</servlet-name>
-     <servlet-class>org.eclipse.jetty.servlet.DefaultServlet</servlet-class>
-     <init-param>
-       <param-name>useFileMappedBuffer</param-name>
-       <param-value>false</param-value>
-     </init-param>
-     <load-on-startup>0</load-on-startup>
-   </servlet>
- ...
- </web-app>    
+  <_initParams>
```

Comment:
TIL that `"_"` (underscore) is valid for the start of an XML element name.
Still seems strange.  Why is this not just `<initParams>` ?

Reply:
Because you cannot call a setter that has 2 args, which setInitParamater(String,String) does. Therefore, we are using the field directly, which is prefixed by _.